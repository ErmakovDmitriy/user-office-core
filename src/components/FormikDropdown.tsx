import React from "react";
import { TextField } from "formik-material-ui";
import { MenuItem } from "@material-ui/core";
import { Field } from "formik";

class FormikDropdown extends React.Component<TProps> {
  render() {
    return (
      <Field
        type="text"
        name={this.props.name}
        label={this.props.label}
        select
        margin="normal"
        component={TextField}
        InputLabelProps={{
          shrink: true
        }}
        fullWidth
        required={this.props.required}
        disabled={this.props.disabled}
      >
        {this.props.children}
        {this.props.items.map(option => {
          return (
            <MenuItem key={option.value} value={option.value}>
              {option.text}
            </MenuItem>
          );
        })}
      </Field>
    );
  }
}

interface TProps {
  items: Option[];
  name: string;
  label: string;
  required?: boolean;
  disabled?: boolean;
}

interface Option {
  text: string;
  value: string;
}

export default FormikDropdown;
