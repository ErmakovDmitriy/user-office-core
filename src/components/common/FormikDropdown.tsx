import MenuItem from '@material-ui/core/MenuItem';
import { Field } from 'formik';
import { TextField } from 'formik-material-ui';
import PropTypes from 'prop-types';
import React, { PropsWithChildren } from 'react';

type TProps = {
  items: Option[];
  name: string;
  label: string;
  required?: boolean;
  disabled?: boolean;
};

const FormikDropdown: React.FC<PropsWithChildren<TProps>> = ({
  name,
  label,
  required,
  disabled,
  children,
  items,
}) => {
  return (
    <Field
      type="text"
      name={name}
      label={label}
      select
      margin="normal"
      component={TextField}
      InputLabelProps={{
        shrink: true,
      }}
      fullWidth
      required={required}
      disabled={disabled}
    >
      {children}
      {items.map(option => {
        return (
          <MenuItem key={option.value} value={option.value}>
            {option.text}
          </MenuItem>
        );
      })}
    </Field>
  );
};

export interface Option {
  text: string;
  value: string | number;
}

FormikDropdown.propTypes = {
  items: PropTypes.array.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  required: PropTypes.bool,
  disabled: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default FormikDropdown;
