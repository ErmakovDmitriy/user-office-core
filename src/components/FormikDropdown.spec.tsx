import React from 'react';
import FormikDropdown from './FormikDropdown';
import ReactTestRenderer from 'react-test-renderer';
import { Formik } from "formik";

test('Dropdown populated with given values', () => {
  const renderer = ReactTestRenderer.create(
    <Formik initialValues={{name: "name"}} onSubmit={() => {}}>
    <FormikDropdown 
    name="name" 
    label="label" 
    items={[{ text: "One", value: "one" }, { text: "Two", value: "two" }]} />
    </Formik>
  );

  let tree = renderer.toJSON();
  expect(tree).toMatchSnapshot();
});