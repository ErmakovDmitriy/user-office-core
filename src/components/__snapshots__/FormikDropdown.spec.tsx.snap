// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown populated with given values 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
  >
    label
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="true"
      aria-pressed="false"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      id="select-name"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      name="name"
      type="hidden"
      value="name"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
